thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/wrapper/ChildKeepingMergeWrapper.java

```diff
@@ -97,10 +106,11 @@ public ChildKeepingMergeWrapper(DataProvider provider, DetectionConfigDTO config
       }
     }
 
-    // refill current and baseline values for qualified parent anomalies
+    // Refill current and baseline values for qualified parent anomalies
+    // Ignore filling baselines for exiting parent anomalies and grouped anomalies
     Collection<MergedAnomalyResultDTO> parentAnomalies = Collections2.filter(output,
         mergedAnomaly -> mergedAnomaly != null && !mergedAnomaly.getChildren().isEmpty() && !isExistingAnomaly(
-            existingParentAnomalies, mergedAnomaly));
+            existingParentAnomalies, mergedAnomaly) && !StringUtils.isBlank(mergedAnomaly.getMetric()));
```

Comment:
You can fill current and baseline values only on anomalies having a metric name. This condition basically filters out parent anomalies not having a metric. Of course, grouped anomalies do not have a metric name but I wouldn't consider that as the source of truth to label a grouped anomaly because there may be other types of anomalies in the future without metric names.

Reply:
Make sense.